packages/core/src/storage/Metadata.ts

```diff
@@ -0,0 +1,79 @@
+export type MetadataBase = {
+  [key: string]: Record<string, unknown>
+}
+
+/**
+ * Metadata access class to get, set (create and update) and delete
+ * metadata on any record.
+ *
+ * set will override the previous value if it already exists
+ *
+ * note: To add in-memory persistence to these records, you have to
+ * update the record in the correct repository
```

Comment:
Dumb confirmation here.. So you essentially _do need_ to store the altered record in the associated record repository.. but that shouldn't require changing any types associated with that record, right?

Reply:
Yes! 

'''ts
connectionRecordExample.metadata.set('test', { foo : 'bar' } );

connectionRepository.update(connectionRecordExample);
'''

I was unable to automatically store the record, so I think that this is the best solution.